cpp/include/cudf/io_types.h

```diff
@@ -133,12 +133,27 @@ typedef struct {
 
   char          *encoding;                  // the data encoding, NULL = UTF-8
 
-  size_t        byte_range_offset;          ///< offset of the byte range to read. 
+  size_t        byte_range_offset;          ///< offset of the byte range to read.
   size_t        byte_range_size;            /**< size of the byte range to read. Set to zero to read all data after byte_range_offset.
                                             Reads the row that starts before or at the end of the range, even if it ends after the end of the range. */
 } csv_read_arg;
 
 
+typedef struct
+{
+    gdf_column** data;            // columns to output
+    int num_cols;                 // number of columns
+
+    const char* filepath;         // full path to file to create
+    bool windows_line;            // use \r\n for line terminator; ignore line_terminator char
+    char line_terminator;         // character to use for separating lines (default '\n')
```

Comment:
Is this modeled after pandas' interface?  If not I would probably merge `windows_line` and `line_terminator` into a single parameter of type `const char *` with default `"\n"`.